clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -756,18 +846,18 @@ public void testFencedMemberStopHeartbeatUntilItReleasesAssignmentToRejoin() {
     @ParameterizedTest
     @ApiKeyVersionsSource(apiKey = ApiKeys.CONSUMER_GROUP_HEARTBEAT)
     public void testSendingLeaveGroupHeartbeatWhenPreviousOneInFlight(final short version) {
-        mockStableMember();
         time.sleep(DEFAULT_HEARTBEAT_INTERVAL_MS);
         NetworkClientDelegate.PollResult result = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(1, result.unsentRequests.size());
         result = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(0, result.unsentRequests.size(), "No heartbeat should be sent while a previous one is in-flight");
 
-        membershipManager.leaveGroup();
-
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(true);
```

Comment:
we should `when(membershipManager.state()).thenReturn(MemberState.LEAVING)` instead I would say, to be true to the real flow. When a member is leaving the HB is generated because of the state it's in (and it also considers the canSend request, but with the state as a spy do we really need to set this expectation?)

Reply:
You're right, replacing that now, thank you